### ISSUE TITLE
AuthorityConsumer never processes itemsToUpdateAuthority? Or needs documentation?

### ISSUE THREAD

**Issue by kshepherd** [DSpace/DSpace#8209]
In `AuthorityConsumer` there are two lists of items that are processed for reindexing, but one of the lists is either redundant, or just broken, or both? From what I can see, there is even item event detail containing 'ARCHIVED', unless this is populated by some method I don't know about, outside of configuration or code?

The following check determines whether to add the item to the list:

```
            if(("ARCHIVED: " + true).equals(event.getDetail())){
                itemsToUpdateAuthority.add(item.getID());
            }
```
Is this something we need to fix, or is it a redundancy we can simplify and remove?

**Comment by tdonohue** [comment]
@kshepherd : Yes, they look redundant to me too... this seems similar to #2990 in that it's in the Authority code, which is very poorly commented overall.

Based on a glance at the code though, you look correct.  
1. There's two `Set<UUID>` initialized here, with seemingly different purposes: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authority/indexer/AuthorityConsumer.java#L38-L46
2. However, they both seem to be populated when the Item is "archived" here: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authority/indexer/AuthorityConsumer.java#L69-L77  (Though these do slightly different checks, so we may want to double check the those checks are equivalent)
3. Then, regardless of which set is populated, they both trigger the same reindex here: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authority/indexer/AuthorityConsumer.java#L90-L99

So, it does look like we may be triggering a reindex twice when an Item is archived.  That said, we may want to ensure that the checks in (2) are truly identical (they seem to be at a glance, but I haven't dug deeply enough).
